src/Validation/ValidatorAwareTrait.php

```diff
@@ -95,8 +95,16 @@ public function validator($name = null, Validator $validator = null)
         }
 
         if ($validator === null) {
-            $validator = new Validator();
-            $validator = $this->{'validation' . ucfirst($name)}($validator);
+            $className = isset($this->_validatorClass) ? $this->_validatorClass : 'Validator';
+            if ($className === 'Validator') {
+                $className = '\Cake\Validation\Validator';
+            }
+            $className = App::className($className, 'Validation/Validator', 'Validator');
+            if (!$className) {
+                throw new Exception\MissingValidatorException(['class' => $className]);
+            }
+
+            $validator = $this->{'validation' . ucfirst($name)}(new $className);
```

Comment:
In the past I've handled this by returning a new validator from the validation hook method. Are you wanting to have a custom validator class for _all_ the validators in a table?